src/Statistics.cpp

```diff
@@ -215,6 +217,36 @@ read(const std::string& path)
   return counters;
 }
 
+Counters
+read_log(const std::string& path)
+{
+  Counters counters(true);
+
+  std::string data;
+  data = Util::read_file(path);
+
+  std::stringstream ss(data);
+  std::string line;
+  while (getline(ss, line, '\n')) {
+    if (line[0] == '#') {
+      continue;
+    }
+    Statistic statistic = Statistic::none;
+    for (const auto& field : k_statistics_fields) {
```

Comment:
Not sure if it's a major problem in practice, but it would be nice to avoid having to loop once per logged result.

Here's an idea: What if we change the log format to something like

```
# direct_cache_hit path/to/file.c
22
```

or

```
# path/to/file.c
22 # direct_cache_hit
```

(where 22 is the index in `Counters` for `direct_cache_hit`) instead and then call `counters.increment_raw(index)` for each logged result?

Reply:
This is just a minor thing; I'm OK with keeping the current format. Perhaps it's even better not to leak implementation details like array index into the log.

Reply:
We can make a lookup table for the parsing, the idea was to keep the stats log human readable (and version-proof)

But the number is probably readable anyway, and as long as you don't remove indexes it would be "safe" as well.

It would be possible to remove the file name and write only the result code (like in the `stats` files, their order)

Maybe will try with some bigger builds first, and see how big the stats log gets for those. Like, if it is a problem ?

Reply:
Right, let's leave it as is and maybe optimize it with a lookup table later if needed. :+1: